astropy/config/paths.py

```diff
@@ -7,76 +7,12 @@
 import shutil
 import sys
 from functools import wraps
+from pathlib import Path
 
 __all__ = ["get_config_dir", "get_cache_dir", "set_temp_config", "set_temp_cache"]
 
 
-def _find_home():
-    """Locates and return the home directory (or best approximation) on this
-    system.
-
-    Raises
-    ------
-    OSError
-        If the home directory cannot be located - usually means you are running
-        Astropy on some obscure platform that doesn't have standard home
-        directories.
-    """
-    try:
-        homedir = os.path.expanduser("~")
-    except Exception:
-        # Linux, Unix, AIX, OS X
-        if os.name == "posix":
-            if "HOME" in os.environ:
-                homedir = os.environ["HOME"]
-            else:
-                raise OSError(
-                    "Could not find unix home directory to search for "
-                    "astropy config dir"
-                )
-        elif os.name == "nt":  # This is for all modern Windows (NT or after)
-            if "MSYSTEM" in os.environ and os.environ.get("HOME"):
-                # Likely using an msys shell; use whatever it is using for its
-                # $HOME directory
-                homedir = os.environ["HOME"]
-            # See if there's a local home
-            elif "HOMEDRIVE" in os.environ and "HOMEPATH" in os.environ:
-                homedir = os.path.join(os.environ["HOMEDRIVE"], os.environ["HOMEPATH"])
-            # Maybe a user profile?
-            elif "USERPROFILE" in os.environ:
-                homedir = os.path.join(os.environ["USERPROFILE"])
-            else:
-                try:
-                    import winreg as wreg
-
-                    shell_folders = r"Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders"
-                    key = wreg.OpenKey(wreg.HKEY_CURRENT_USER, shell_folders)
-
-                    homedir = wreg.QueryValueEx(key, "Personal")[0]
-                    key.Close()
-                except Exception:
-                    # As a final possible resort, see if HOME is present
-                    if "HOME" in os.environ:
-                        homedir = os.environ["HOME"]
-                    else:
-                        raise OSError(
-                            "Could not find windows home directory to "
-                            "search for astropy config dir"
-                        )
-        else:
-            # for other platforms, try HOME, although it probably isn't there
-            if "HOME" in os.environ:
-                homedir = os.environ["HOME"]
-            else:
-                raise OSError(
-                    "Could not find a home directory to search for "
-                    "astropy config dir - are you on an unsupported "
-                    "platform?"
-                )
-    return homedir
-
-
-def get_config_dir(rootname="astropy"):
+def get_config_dir(rootname: str = "astropy") -> str:
```

Comment:
Followup PR to return a Path object?

Reply:
I depends wether this API is considered public or private: if it's private, I can do it now, and if it's public, we probably shouldn't do it at all. In doubt, I erred on the side of caution.

Reply:
It looks to be public. So followup?
I think it's a very good thing to return Path objects, not string, when dealing with paths. Backwards compatibility is very easy for downstream libraries: just call `str`. But probably they want Paths long term :).

Reply:
Dully noted. I'll circle back to this.

Reply:
Let's not break downstream just because ruff is complaining here...